src/CatBox.NET/Throw.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace CatBox.NET;
+
+/// <summary>
+/// This library targets .NET Standard 2.1 and thus, does not contain the newer ArgumentException family of guard clauses
+/// This is a backport of those two methods.
+/// </summary>
+internal static class Throw
+{
+    public static void IfStringIsNullOrWhitespace([DoesNotReturnIf(true)] string? s, string exceptionMessage, [CallerArgumentExpression("s")] string memberName = "")
+    {
+        if (string.IsNullOrWhiteSpace(s)) throw new ArgumentNullException(memberName, exceptionMessage);
+    }
+
+    public static void IfNull([DoesNotReturnIf(true)] object? s, [CallerArgumentExpression("s")] string memberName = "")
+    {
+        if (s is null) throw new ArgumentNullException(memberName, "Argument cannot be null");
+    }
```

Comment:
Since the library includes a minimum target to .NET Standard 2.1 (the minimum version for C# 8 to support async-enumerables), the `ArgumentNullException.ThrowIfNull` isn't present in the BCL at this point. That was added in a later version of .NET